rollup-operator/src/PoB/synch-pob.js

```diff
@@ -0,0 +1,481 @@
+        this.contractPoB = new this.web3.eth.Contract(rollupPoBABI, this.rollupPoBAddress, burnAddress, {handleRevert: true});
+        this.winners = [];
+        this.currentWinners = [];
+        this.slots = [];
+        this.bids = [];
+
+        this._initTimeouts(timeouts);
+        this._initLogger(logLevel);
+    }
+
+    /**
+     * Initilaize all timeouts
+     * @param {Object} timeouts 
+     */
+    _initTimeouts(timeouts){
+        const errorDefault = 5000;
+        const nextLoopDefault = 5000;
+        const loggerDefault = 5000;
+
+        let timeoutError = errorDefault;
+        let timeoutNextLoop = nextLoopDefault;
+        let timeoutLogger = loggerDefault;
+
+        if (timeouts !== undefined) {
+            timeoutError = timeouts.ERROR || errorDefault;
+            timeoutNextLoop = timeouts.NEXT_LOOP || nextLoopDefault;
+            timeoutLogger = timeouts.LOGGER || loggerDefault;
+        }
+
+        this.timeouts = {
+            ERROR: timeoutError,
+            NEXT_LOOP: timeoutNextLoop,
+            LOGGER: timeoutLogger,
+        };
+    }
+
+    /**
+     * Initilaize logger
+     * @param {String} logLevel 
+     */
+    _initLogger(logLevel) {
+        // config winston
+        var options = {
+            console: {
+                level: logLevel,
+                format: winston.format.combine(
+                    winston.format.colorize(),
+                    winston.format.simple(),
+                )
+            },
+        };
+
+        this.logger = winston.createLogger({
+            transports: [
+                new winston.transports.Console(options.console)
+            ]
+        });
+    }
+
+    /**
+     * Convert to string
+     * normally used in order to add it to database
+     * @param {Any} - any input parameter
+     * @returns {String}
+     */
+    _toString(val) {
+        return JSON.stringify(stringifyBigInts(val));
+    }
+
+    /**
+     * Get from string
+     * normally used ti get from database
+     * @param {String} - string to parse
+     * @returns {Any} 
+     */
+    _fromString(val) {
+        return unstringifyBigInts(JSON.parse(val));
+    }
+
+    /**
+     * Get public contract variables
+     * Initialize array winners/slots
+     * Start logger
+     */
+    async _init(){
+        // Initialize class variables
+        this.genesisBlock = 0;
+        this.totalSynch = 0;
+        this.blocksPerSlot = Number(await this.contractPoB.methods.BLOCKS_PER_SLOT()
+            .call({from: this.ethAddress}));
+        this.slotDeadline = Number(await this.contractPoB.methods.SLOT_DEADLINE()
+            .call({from: this.ethAddress}));
+        this.maxTx = Number(await this.contractPoB.methods.MAX_TX()
+            .call({from: this.ethAddress}));
+        this.minBid = Number(await this.contractPoB.methods.MIN_BID()
+            .call({from: this.ethAddress}));
+        this.blocksNextInfo = this.blocksPerSlot;
```

Comment:
I think it makes the code more readable since `this.blocksNextInfo` is used for loop when synchronize and `this.blocksPerSlot` is used when polling the synchronizer Api